File: AutomaticBrewingCoffeeApp/lib/app/data/enums/payment_gateway.dart
enum PaymentGateway {
  vnpay,
  reso,
  mpos,
  atm;
  
  String get value {
    switch (this) {
      case PaymentGateway.vnpay:
        return 'VNPay';
      case PaymentGateway.mpos:
        return 'MPOS';
      case PaymentGateway.reso:
        return 'Reso';
      case PaymentGateway.atm:
        return 'ATM';
    }
  }
  
  static PaymentGateway fromString(String value) {
    switch (value.toLowerCase()) {
      case 'vnpay':
        return PaymentGateway.vnpay;
      case 'mpos':
        return PaymentGateway.mpos;
      case 'reso':
        return PaymentGateway.reso;
      case 'atm':
        return PaymentGateway.atm;
      default:
        return PaymentGateway.vnpay; // Default value
    }
  }
}

File: AutomaticBrewingCoffeeKioskBE/ABC_Kiosk_BE/Domain/Enums/PaymentGateway.cs
using System.ComponentModel.DataAnnotations;

namespace Domain.Enums;

public enum PaymentGateway
{
    //[Display(Name = "VNPay")] VNPay = 0,

    //[Display(Name = "MoMo")] MoMo = 1
    [Display(Name = "MPOS")] MPOS = 1

}